apps/backend/src/groups/groups.service.ts

```diff
@@ -58,19 +61,47 @@ export class GroupsService {
     const groupUser = await GroupUser.findOne({
       where: {groupId: group.id, userId: updateGroupUser.userId}
     });
+    if (groupUser) {
+      await this.setDefaultToOwner(group, groupUser.userId);
+    }
     return groupUser?.update({role: updateGroupUser.groupRole});
   }
 
   async removeUserFromGroup(group: Group, user: User): Promise<Group> {
+    await this.setDefaultToOwner(group, user.id);
+    return group.$remove('user', user);
+  }
+
+  async setDefaultToOwner(group: Group, id: string) {
```

Comment:
'id' of what?  i would do 'userId' here if forced to only have the attribute, but it seems like the current practice is to pass through these model objects in their entirety and then use attributes from them as necessary, so i would actually probably make this 'user' and then change the conditional on line 80 to get the attribute